### PR TITLE
DDFLSBP-651 - Implement cookie blocking placeholder

### PR DESCRIPTION
**Pull request by Dresse**
#### Link to issue

https://reload.atlassian.net/browse/DDFLSBP-651

#### Description

This PR introduces the following:

- A new custom module (dpl_cookies), used for handling all custom cookie functionality, that is not handled in either the cookieinformation module or cookieinformation dashboard. For now this includes a preprocess hook that adds 3 new data attributes to all video media elements. These attributes are need by the cookieinformation script when handling user cookie consent. 

- A new template specifically targeting field--media--video. In this template we have added our custom cookie placeholder, which gets shown to user who have not yet consented to using cookies, in place of the original content.

#### Screenshot of the result
Screenrecording showing the result:

https://github.com/danskernesdigitalebibliotek/dpl-cms/assets/6142323/def4e7d4-884b-448c-8130-21b7b567da59

#### Additional comments or questions

This PR has a sister PR in the dpl-design-system: https://github.com/danskernesdigitalebibliotek/dpl-design-system/pull/650

### CODE REVIEW

File: web/themes/custom/novel/templates/fields/field--media--video.html.twig [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  <div class="cookie-placeholder__wrapper">
+    <div class="cookie-placeholder__description">
+        <div id="dpl-react-apps-cookie-placeholder">
+        {{ "To view this content, we need your consent to use cookies."|t({}, {'context': 'Cookie Placeholder'}) }}
+        </div>
+    </div>
+    <button
+        type="button"
+        class="cookie-placeholder__manage-consent-button btn-primary btn-outline btn-xlarge arrow__hover--right-small"
+        aria-label="Manage consent"
```
Comment (clausbruun) [inline web/themes/custom/novel/templates/fields/field--media--video.html.twig:15]:
```suggestion
        aria-label="{{ 'Manage consent'|t({}, {'context': 'Cookie Placeholder'}) }}"
```

Comment (Dresse) [reply web/themes/custom/novel/templates/fields/field--media--video.html.twig]: Good find! Added it now. 👍

File: web/modules/custom/dpl_cookies/dpl_cookies.module [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+/**
+ * @file
+ * DPL cookies module.
+ */
+
+/**
+ * Implements hook_preprocess_media().
+ */
```
[the remark this answers is not in the document]
Comment (Dresse) [reply web/modules/custom/dpl_cookies/dpl_cookies.module]: Added suggested comment explaining the preprocess function. 👍

**Comment by Dresse** [comment]
> Hvis jeg afviser alle cookies ser jeg dette på en side hvor der er indsat et Vimeo klip:
> 
> Skærmbillede 2024-06-05 kl 11 42 19 Det er muligvis mit lokalmiljø som driller med stylingen, men under alle omstændigheder kunne det måske være en forbedring at lave `<div class="video-embed__wrapper">` elementet lidt lavere, når `cookie-placeholder` vises, således at der ikke er så meget "tom" plads?

Good suggestion. I have already used some time on this PR, so I think I will skip this for now. Maybe this is also something that we need to clear up with the client beforehand. 😄

**Review by clausbruun — COMMENTED** [review]
LGTM 🎉

**Pull request merged**